website/hw/hw4.rst

```diff
@@ -180,103 +218,389 @@ As you are stepping through your program, you might want to be able to
 see the values of certain variables. You can do this by running the
 ``print`` command. ``print val``, for example, will print the value
 ``val``. With pointers, you can use ``print *p`` to print out the value
-pointed to by p. If you want to print in hex, use ``print/x val``, and
+pointed to by p. If you want to print in hex, use ``print/x val``, and
 to print in binary use ``print/b val``
 
 There is also a useful command to keep track of when a variable changes.
-This command is ``watch``. If you want to keep track of when the
+This command is ``watch`` (careful: you can't use this command before
+you start running the program in GDB; otherwise, GDB won't know what
+variables are currently in scope). If you want to keep track of when the
 variable ``x`` changes, type ``watch x``. Every time it is changed when
 you run your program or execute a line of code, the program will stop
-and print the old and new values.
+and print the old and new values (note that it will break in the line
+*after* the variable was changed).
 
 When you’re done using gdb, you can exit by typing ``quit`` or by
 pressing Ctrl+D.
```

Comment:
Right now this doesn't note that they'll get the following notice:
```
A debugging session is active.

	Inferior 1 [process 642037] will be killed.

Quit anyway? (y or n)
```
it may be useful to mention